lib/pbench/server/api/__init__.py

```diff
@@ -70,6 +72,55 @@ def create_app():
     return app
 
 
+def _build_cors_preflight_response():
+    response = make_response()
+    response.headers.add("Access-Control-Allow-Origin", "*")
+    response.headers.add("Access-Control-Allow-Headers", "*")
+    response.headers.add("Access-Control-Allow-Methods", "*")
+    return response
+
+
+def _corsify_actual_response(response):
+    response.headers.add("Access-Control-Allow-Origin", "*")
+    return response
+
+
+class Download(Resource):
+    def post(self):
+        global app
+        elasticsearch = "http://elasticsearch.perf.lab.eng.bos.redhat.com:9280"
+        try:
+            # query ElasticSearch
+            json_data = request.get_json()
+            if "params" in json_data:
+                url = f"{elasticsearch}{json_data['url']}?{json_data['params']}"
+            else:
+                url = f"{elasticsearch}{json_data['url']}"
+            if "payload" in json_data:
+                response = requests.post(url, json=json_data["payload"])
+            else:
+                response = requests.get(url)
+
+            # construct response object
+            response = make_response(response.text)
+            response = _corsify_actual_response(response)
+        except Exception:
+            app.logger.debug("There was something wrong with the POST request.")
```

Comment:
Hi, Is it possible if we can add more details why the post request failed, maybe we can log the response code like 500, or 401/403. That will help us debug later. This quickstart might help in writing exceptions https://requests.readthedocs.io/en/latest/user/quickstart/#errors-and-exceptions

Reply:
Perhaps substitute `debug` with `exception`.